content/post/2024-10-01-same-same-but-different-how-advanced-data-science-techniques-help-us-validate-drug-names/index.md

```diff
@@ -0,0 +1,628 @@
+##  [70] "perphenazine"                                  
+##  [71] "pethidine"                                     
+##  [72] "phenylephrine"                                 
+##  [73] "pipotiazine"                                   
+##  [74] "pirprofen"                                     
+##  [75] "plerixafor"                                    
+##  [76] "potassium citrate"                             
+##  [77] "prazosin"                                      
+##  [78] "prednisone"                                    
+##  [79] "remoxipride"                                   
+##  [80] "reteplase"                                     
+##  [81] "rifamycin"                                     
+##  [82] "rivastigmine"                                  
+##  [83] "roxithromycin"                                 
+##  [84] "salsalate"                                     
+##  [85] "sorbitol"                                      
+##  [86] "streptokinase"                                 
+##  [87] "succinimide"                                   
+##  [88] "taurolidine"                                   
+##  [89] "technetium (99mTc) pertechnetate"              
+##  [90] "teneligliptin"                                 
+##  [91] "theophylline, combinations excl. psycholeptics"
+##  [92] "ticarcillin"                                   
+##  [93] "tiemonium iodide and analgesics"               
+##  [94] "timolol, thiazides and other diuretics"        
+##  [95] "tolperisone"                                   
+##  [96] "tramadol"                                      
+##  [97] "tretoquinol"                                   
+##  [98] "trypsin, combinations"                         
+##  [99] "ursodoxicoltaurine"                            
+## [100] "zidovudine"
+```
+
+We're looking to add a bit of confusion to our drug names, so we've created a function called `introduce_variation`. It takes a name and returns a new version with a duplicate, deleted, or rearranged character.
+
+
+``` r
+introduce_variation <- function(name) {
+  # Randomly choose a type of modification to introduce a typo
+  modification <- sample(c("duplicate", "remove", "swap"), 1)
+  name_chars <- unlist(strsplit(name, ""))
+  
+  if (modification == "duplicate") {
+    # Duplicate a random character
+    duplicate_pos <- sample(1:length(name_chars), 1)
+    name_chars <- append(name_chars, name_chars[duplicate_pos], after = duplicate_pos)
+  } else if (modification == "remove") {
+    # Remove a random character
+    remove_pos <- sample(1:length(name_chars), 1)
+    name_chars <- name_chars[-remove_pos]
+  } else if (modification == "swap") {
+    # Swap two adjacent characters
+    swap_pos <- sample(1:(length(name_chars) - 1), 1)
+    temp <- name_chars[swap_pos]
+    name_chars[swap_pos] <- name_chars[swap_pos + 1]
+    name_chars[swap_pos + 1] <- temp
+  }
+    
+  return(paste(name_chars, collapse = ""))
+}
+```
```

Comment:
The introduce_variation function adds complexity that might not be necessary for this article. Since the focus is on fixing errors in drug names, introducing random typos as part of the workflow could confuse readers. Pre-preparing a small set of intentional errors and using them consistently would simplify the explanation and keep the focus on error correction rather than error creation.